### ISSUE TITLE
FEATURE REQUEST maximize & restore Vim & Neovim splits while inside a tmux pane ...please ...& thank you

### ISSUE THREAD

**Issue by ipatch** [christoomey/vim-tmux-navigator#227]
Hi, 👋 been using this plugin for some time now, absolutely couldn't work without it.  Kudos 🍫 to [all the contributors](https://github.com/christoomey/vim-tmux-navigator/graphs/contributors) of the plugin. That said, I tend to maximize and minimize tmux panes inside a tmux window on a regular basis, most notably when I'm working with a project that requires the following:

- a pane for a text editor 
- running a server of some sort 
- and a pane for inputting commands, _often git related._

I can usually get by with quickly switching between buffers in my text editor tmux pane, but there are occasions where I open a split in Vim or Neovim, and it's often a vertical split and I'm working with a react based component file, and also prefer to have a separate CSS file styling the component open in the same tmux pane side by side vertically.  vim-tmux-navigator works great for quickly jumping around the splits 🤸‍♀️ in Vim and the panes in tmux, however maximizing a split inside of Vim or Neovim and then navigating with tmux navigator is when things don't go so well.

In my use case, I like to maximize my text editor pane for obvious reasons, ie. working with lines that are longer than the usual 80 chars, when I have two vertical splits open, etc, etc.  I think it would make for a great feature if vim-tmux-navigator could maximize and then restore splits within Vim & Neovim using the standard key binding of <kbd>prefix</kbd> + <kbd>z</kbd> used within tmux to quickly maximize & restore splits in Vim and Neovim as it seamlessly navigates between Vim splits and tmux panes.

All that said, not exactly sure what the current state of project affairs are, as I see two active PR's open, and haven't seen a commit merged or pushed into master since late of last year, so I thought I'd leave this FR here in hopes of seeing if we could get a dialogue going about this feature.

Please excuse my long winded rant, as I'd rather type too much than not enough.

**Comment by christoomey** [comment]
Hi @ipatch, thanks for sharing! To first answer the "current state of project affairs" question, this project is very much still alive, but I'm mostly considering it in maintenance mode and tend to be very cautious when considering adding any new functionality, thus the lack of commits in the last year. My main priority is getting as close to "it just works" as possible, so any new additions are measured through that lens.

As to your suggestion, if I'm understanding correctly, you're saying that you'd like the `prefix-z` combo to maximize and minimize Vim splits when the Vim pane is maximized in tmux. I may be missing something, but I think that would break the default functionality where `prefix-z` toggles the zoom state in tmux. As such, I don't consider it safe to override that key functionality and as such wouldn't want to implement this in the repo.

That said, for this specific case you thankfully aren't blocked at all by the plugin and could implement this entierly outside vim-tmux-navigator. I believe if would look something like:

``` tmux
# normal tmux navigator is_vim check
is_vim="ps -o state= -o comm= -t '#{pane_tty}' \
    | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?(view|n?vim?x?)(diff)?$'"

# new binding for prefix-z, sending <Ctrl-q> to Vim as a way
# to communicate
bind-key z if-shell "$is_vim" "send-keys C-q"  "resize-pane -Z"
```

``` vim
function! s:TogglePaneZoom()
  " ... releavant Vim toggle code here
endfunction

nnoremap <C-q> call s:TogglePaneZoom()<cr>
```

That should provide a rough scaffold if you want to dig deeper into this. For the reasons stated above, I don't believe I'd want to integrate that into the plugin as I'm strongly biased towards not altering default behavior, so I'm going to close this now, but please feel free to comment if misunderstood.

**Comment by ipatch** [comment]
Thanks for scaffolding out a possible solution, I'll totally mess around with it some more, and see what I can figure out.  And if I can come up with something tangible I'll try and link an asciicast / youtube video to better describe the feature I'd like to see, and if possible I'll post what I ended settling with.  Thanks again, for the heads, and for making a great plugin.

cheers 🍻

**Comment by timblaktu** [comment]
I work similarly to @ipatch here (and have the same troubles). Was wondering if @ipatch came to some acceptable solution?

Although I also use a combination of tmux panes and vim splits (which is of course, why we're all here! Thanks @christoomey! ;-) ) I would rephrase the use case/need just a bit:

* A tmux pane running vim is maximized (using `prefix + z` )
** this should be a very common use case indeed; screen real estate is a hot commodity for sw devs
* vim has multiple windows/splits

My _need_ in this case is for my navigation bindings NOT to pop me out of my maximized vim view. I do this accidentally all the time, and have to go back to vim and `prefix + z`. 

I believe the solution to this would be a modification to the conditions under which `select-pane` command is issued, in [the suggested config snippet](https://github.com/christoomey/vim-tmux-navigator#add-a-snippet).

I just wanted to propose this idea here first, before taking a stab at it myself, in case someone's already done something like this.

I just upgraded yesterday from 3 yr old versions of tmux/vim/etc and have found a lot of things have changed WRT configuration. I don't know the full extent, but things started working once I replaced what I had been doing in my tmux.conf/.vimrc to using [the suggested config snippet](https://github.com/christoomey/vim-tmux-navigator#add-a-snippet). However, I still have this essential problem stated above.
